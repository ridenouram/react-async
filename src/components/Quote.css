.Quote {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  padding: 30px;
  margin: 10px;
  width: 400px;
  border: black 2px;
  background: salmon;
  h3 {
    padding: 10px;
  }
  img {
    width: 200px;
  }
  p {
    background: rgb(233, 188, 171);
    padding: 10px;
  }
}

.Quotes {
  display: flex;
  flex-wrap: wrap;
}
