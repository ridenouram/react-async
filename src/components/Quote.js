import React from 'react';
import PropTypes from 'prop-types';

export default function Quote({ character, image, quote }) {
  return (
    <container style={{ border: 'black 3px' }}>
      <h3>{character}</h3>
      <img src={image} />
      <p>{quote}</p>
    </container>
  );
}

Quote.propTypes = {
  character: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  quote: PropTypes.string.isRequired
};
