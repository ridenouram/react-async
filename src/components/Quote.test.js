import React from 'react';
import { shallow } from 'enzyme';
import Quote from './Quote';

describe('Quote component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<Quote character='name' image='img.jpg' quote='quotey mcquote' />);
    expect(wrapper).toMatchSnapshot();
  });
});
