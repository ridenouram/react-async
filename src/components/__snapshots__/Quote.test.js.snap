// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component matches snapshot 1`] = `
<container
  className="Quote"
>
  <h3 />
  <img />
  <p>
    quotey mcquote
  </p>
</container>
`;
